treeherder/auth/backends.py

```diff
@@ -99,32 +100,30 @@ def authenticate(self, auth_header=None, host=None, port=None):
 
         if result["status"] != "auth-success":
             logger.warning("Error logging in: {}".format(result["message"]))
-            raise TaskclusterAuthenticationFailed(result["message"])
-
-        # TODO: remove this size limit when we upgrade to django 1.10
-        # in Bug 1311967
-        email = self._get_email(result)
-        if len(email) <= 30:
-            username = email
-        else:
-            username = base64.urlsafe_b64encode(
-                hashlib.sha1(smart_bytes(email)).digest()
-                ).rstrip(b'=')[-30:]
+            raise TaskclusterAuthException(result["message"])
 
+        client_id = result["clientId"]
+        email = self._extract_email_from_clientid(client_id)
+
+        # Look for an existing user in this order:
+        #
+        # 1. Matching username/clientId
+        # 2. Matching email.
+        # Otherwise, create it, as long as it has an email.
         try:
-            # Find the user by their email.
-            user = self._get_user(email)
-            user.username = username
+            return User.objects.get(username=client_id)
 
         except ObjectDoesNotExist:
-            # the user doesn't already exist, create it.
-            logger.warning("Creating new user: {}".format(username))
-            user = User(email=email,
-                        username=username,
-                        )
+            try:
+                # TODO: remove this once all users are converted to clientId
```

Comment:
Ah meant to say - could you file a follow-on bug which plans to remove this at some later point?